service/src/main/java/bio/terra/workspace/service/folder/FolderService.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.workspace.service.folder;
+
+import bio.terra.workspace.db.FolderDao;
+import bio.terra.workspace.db.exception.FolderNotFoundException;
+import bio.terra.workspace.service.folder.model.Folder;
+import com.google.common.collect.ImmutableList;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FolderService {
+
+  private final FolderDao folderDao;
+
+  public FolderService(FolderDao folderDao) {
+    this.folderDao = folderDao;
+  }
+
+  public Folder createFolder(Folder folder) {
+    return folderDao.createFolder(folder);
+  }
+
+  public Folder updateFolder(
+      UUID workspaceUuid,
+      UUID folderId,
+      @Nullable String displayName,
+      @Nullable String description,
+      @Nullable UUID parentFolderId,
+      @Nullable boolean updateParent) {
+    folderDao.updateFolder(
+        workspaceUuid, folderId, displayName, description, parentFolderId, updateParent);
+    return folderDao.getFolder(workspaceUuid, folderId);
+  }
+
+  public Folder getFolder(UUID workspaceUuid, UUID folderId) {
+    return folderDao
+        .getFolderIfExists(workspaceUuid, folderId)
+        .orElseThrow(
```

Comment:
Dao already throws FolderNotFoundException for this case, delete

Though things are bit inconsistent. For getFolder, dao returns FolderNotFoundException. For deleteFolder, service returns FolderNotFoundException. Can we be consistent?

Reply:
yeah.. i don't know what's the rule here. getFolder throw in dao because the sql query throws exception and we wrap it. For delete, there's no exception being thrown.